src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineParser.java

```diff
@@ -157,20 +189,56 @@ public CommandLineParser(final Object callerArguments) {
                 handlePositionalArgumentAnnotation(field, callerArguments);
             }
             if (field.getAnnotation(Argument.class) != null) {
-                handleArgumentAnnotation(field, callerArguments);
+                handleArgumentAnnotation(field, callerArguments, controllingDescriptor);
             }
             if (field.getAnnotation(ArgumentCollection.class) != null) {
                 try {
                     field.setAccessible(true);
-                    createArgumentDefinitions(field.get(callerArguments));
+                    createArgumentDefinitions(field.get(callerArguments), controllingDescriptor);
                 } catch (final IllegalAccessException e) {
                     throw new GATKException.ShouldNeverReachHereException("should never reach here because we setAccessible(true)", e);
                 }
             }
         }
-        return null;
     }
 
+    // Find all the instances of plugins specified by the provided plugin descriptors
+    private void createCommandLinePluginArgumentDefinitions(
+            final List<? extends GATKCommandLinePluginDescriptor<?>> requestedPluginDescriptors) {
+        // For each descriptor, create an argument definition for the descriptor itself, then
```

Comment:
At first I thought not, but I see your point; there can be multiple defs for the descriptor itself. Updated.